### ISSUE TITLE
app/publisher: do not show all published nodes from published root

### ISSUE THREAD

**Issue by mpl** [camlistore/camlistore#882]
If /publishRoot/ is my publisher app handler prefix, and I have published two nodes as "foo", and "bar", the result is that:

http://host:port/publishRoot/foo and http://host:port/publishRoot/bar , 
 
are publicly visible (if known).

However, if I advertise (say on social media), http://host:port/publishRoot/foo/-/hb4e1765b2b , I don't think it's anybody's business to be able to figure out that I also published http://host:port/publishRoot/bar , just by looking at http://host:port/publishRoot/ and see all the published nodes.

I mean, I guess with enough effort they could find out about http://host:port/publishRoot/bar anyway, but no need to make it easier than it has to be?

**Comment by mpl** [comment]
https://camlistore-review.googlesource.com/8866

**Comment by cliveb** [comment]
Agreed. Nice job. Why make it easier. 

Segue, any plans @bradfitz will share roadmap for next Camlistore release focused on Sharing?

**Comment by sllawap** [comment]
mpl@ this change seems odd to me as there is already a mechanism for sharing individual files with the share handler (each of which is gated by a unique via= key. For me personally, the primary use of the publisher is to publish folders instead of individual files and this functionality is now broken.

**Comment by mpl** [comment]
I'm sorry if the change broke something for you, it certainly wasn't meant
to.

Can you please give an example of how it was working for you before and
isn't anymore?

**Comment by sllawap** [comment]
Previously, I was able to setattr camliRoot on a single permanode (a set) and throw images into that set. The images would become instantly visible in the gallery listing and could be downloaded as a zip. This was how I shared albums of images w/ friends and family who were not tech savvy.

I attempted to work around the new 403 behavior on the camliRoot by defining the camliPath:<path> attr on child set permanode, but the contents of that set are not displayed in the publisher index view. I see only an empty gallery.

N.B. It's possible that the secondary failure is due to insecure content being served. I run behind a reverse proxy w/ HTTPS enforced at the edge, and sometimes camlistore/publisher does not honor the proxied schema and tries to outsmart itself. I will verify tomorrow.

**Comment by sllawap** [comment]
Just confirmed: the secondary issue is indeed due to the incorrect address rewriting. Setting camliRoot on a set permanode, and then defining camliPath:xx for a subfolder seems to give me the original functionality.

publisher.js appears to be attempting to fetch http://<servername>/<camliRoot>/search when served from the reverse proxy at https:// which surfaces a "Mixed content" error. I'll file a follow up bug.

**Comment by mpl** [comment]
Coming back to your original issue.

Are you saying that everytime you wanted to share a set, you would set a camliRoot on it, in effect creating a new camliRoot each time? But then you would have to add that new camliRoot to your server config for it to work, wouldn't you?

As you might have guessed I'm a bit puzzled that you managed to make it work that way, as the intended behavior of the publisher is more like to have a unique camliRoot, and from there on to add camliPath children to it.

About the other problem, yes please do open another issue for it.

**Comment by sllawap** [comment]
I had a very simple use for the publisher: a single set permanode which I threw photos into when I wanted to make them public. This seemed to work pretty nicely and once the camliRoot was associated in the server config, I didn't need to make any further modifications. The change to behavior of camliRoot broke my original setup, but the camliPath behavior is sufficient for my needs.

**Comment by sllawap** [comment]
@mpl filed #908 for tracking the mixed content w/ reverse proxy issue.

**Comment by mpl** [comment]
thanks.

Sorry for insisting, but there's still something I don't get about your usage, which I think is important to pin down for future design.

Say you wanted to share photos of event E1 with group of people F1. You would then create permanode P1, and set its camliRoot attribute to "event1".

First question: you then had to add 

	"/someevent/": {
		"camliRoot": "event1",
		...
    }

to the "publish" object of your server configuration, and restart camlistored, hadn't you?

Now, say you wanted to share another event E2 with group of people F2. You would then create P2, with camliRoot = "event2", and again, add

	"/otherevent/": {
		"camliRoot": "event2",
		...
    }

to your config, and restart, correct?

Final question, were you then able to share both events at the same time through "http[s]://yourhost/someevent" and "http[s]://yourhost/otherevent" ?

**Comment by sllawap** [comment]
I only had a single published folder, so I never needed to modify my server config after the initial setup. In that way, I suppose I was not an advanced user! The new mechanism is superior in that it avoids the need to modify the server config when a new published root is added.

The misbehavior I was seeing was conflating the new behavior in this CL (which required some reconfiguration on my part) and #908 which caused the published gallery to appear empty once I had the configuration right. I apologize for the noise!
